components/raftstore/src/store/fsm/peer.rs

```diff
@@ -3653,6 +3721,33 @@ where
     }
 
     fn propose_raft_command(
+        &mut self,
+        msg: RaftCmdRequest,
+        cb: Callback<EK::Snapshot>,
+        diskfullopt: DiskFullOpt,
+    ) {
+        if msg.has_admin_request()
+            && self.fsm.batch_req_builder.request.is_some()
```

Comment:
How can this happen?

Reply:
It's a defense to avoid a case that a admin cmd is proposed and it can change the epoch version but there are some requests in `batch_req_builder ` and their proposed callback are called.
In most of the case, the admin request are proposed through `PeerMsg::RaftCommand` and the `propose_batch_raft_command(true)` will be called so the code here is useless.
The `RollbackMerge` cmd is proposed through calling `propose_raft_command` directly but the `header_checked` in `batch_req_builder` should be false.
In summary, the code here can not be triggered for now.

Reply:
I suggest to remove the code and make `RollbackMerge` to use `propose_batch_raft_command` instead. Or just rename `propose_batch_raft_command` to `propose_raft_command`. Useless code is the source of confusion.